src/main/java/com/jonahseguin/drink/parametric/DrinkProvider.java

```diff
@@ -28,7 +32,7 @@ public T defaultNullValue() {
 
     public abstract String argumentDescription();
 
-    public abstract List<String> getSuggestions(@Nonnull String prefix);
+    public abstract List<String> getSuggestions(@Sender CommandSender sender, @Nonnull String prefix);
```

Comment:
Firstly, the `@Sender` annotation is unnecessary here.
Instead of replacing this method signature entirely, lets try overloading?  to prevent breaking other users' provider implementations and make it backwards-compatible:

```java
public abstract List<String> getSuggestions(@Nonnull String prefix);

// this can be overridden if end users wants access to the sender
public List<String> getSuggestions(CommandSender sender, @Nonnull String prefix) {
    return getSuggestions(prefix);
}
```

Reply:
better yet, we could make these both not abstract and have simple default impl for `getSuggestions(prefix)`:

```java
public List<String> getSuggestions(@Nonnull String prefix) {
    return Collections.emptyList();
}
```